cfme/services/catalogs/catalog_items/__init__.py

```diff
@@ -298,41 +345,86 @@ def delete_if_exists(self):
+        view = navigate_to(self, "Details")
+        path = view.catalog_items.tree.read()
+        path.extend(["Actions", "{} (Group)".format(name)])
+        view.catalog_items.tree.fill(path)
+        view.configuration.item_select("Remove this Button Group", handle_alert=True)
+
+        # ToDo: BZ-1687289 To avoid landing on buttons for catalog item page adding workaround.
+        #  Add flash check and remove browser refresh
+        view.browser.refresh()
+
+    def add_button(self, **kwargs):
+        button_name = kwargs.get("text", "btn_{}".format(fauxfactory.gen_alpha()))
+        hover_name = kwargs.get("hover", "hover_{}".format(fauxfactory.gen_alpha()))
+        image = kwargs.get("image", "broom")
+
         view = navigate_to(self, 'AddButton')
-        if self.appliance.version < '5.9':
-            view.fill({'btn_text': 'btn_text',
-                'btn_hvr_text': button_name,
-                'btn_image': self.button_icon_name,
-                'select_dialog': self.dialog,
-                'system_process': 'Request',
-                'request': 'InspectMe'})
-        else:
-            view.fill({'options': {'btn_text': 'btn_text',
-                                   'btn_hvr_text': button_name,
-                                   'select_dialog': self.dialog,
-                                   'btn_image': self.button_icon_name},
-                       'advanced': {'system_process': 'Request',
-                                    'request': 'InspectMe'}})
+        view.fill({'options': {'btn_text': button_name,
+                               'btn_hvr_text': hover_name,
+                               'select_dialog': self.dialog,
+                               'btn_image': image},
+                   'advanced': {'system_process': 'Request',
+                                'request': 'InspectMe'}})
         view.add.click()
-        view = self.create_view(DetailsCatalogItemView)
-        wait_for(lambda: view.is_displayed, timeout=5)
+        view = self.create_view(DetailsCatalogItemView, wait="10s")
         view.flash.assert_no_error()
         return button_name
 
+    def button_exits(self, name):
+        view = navigate_to(self, "Details")
+        path = view.catalog_items.tree.read()
+        path.extend(["Actions", "{}".format(name)])
+        try:
+            view.catalog_items.tree.fill(path)
+            return True
+        except Exception as e:
+            logger.exception(e)
+            return False
+
+    def delete_button(self, name):
+        view = navigate_to(self, "Details")
+        path = view.catalog_items.tree.read()
+        path.extend(["Actions", "{}".format(name)])
+        view.catalog_items.tree.fill(path)
+        view.configuration.item_select("Remove this Button", handle_alert=True)
+
+        # ToDo: BZ-1687289 To avoid landing on buttons for catalog item page adding workaround.
+        #  Add flash check and remove browser refresh
+        view.browser.refresh()
```

Comment:
Question: This BZ is similar to one I've found in control, is it possible to use a forced navigation rather than a browser refresh? 

Optional: Can you place this in a BZ().blocks statement?